src/App.tsx

```diff
@@ -1,10 +1,61 @@
 /* eslint-disable jsx-a11y/label-has-associated-control */
 /* eslint-disable jsx-a11y/control-has-associated-label */
-import React from 'react';
+import React, { useState, useEffect, useRef } from 'react';
+import classNames from 'classnames';
+
 import { UserWarning } from './UserWarning';
 import { USER_ID } from './api/todos';
+import { Todo } from './types/Todo';
+import { getTodos } from './api/todos';
+
+function filterTodos(todos: Todo[], status?: string | null) {
+  const todosCopy = [...todos];
+
+  if (status === 'active') {
+    return todosCopy.filter(todo => todo.completed === false);
+  }
+
+  if (status === 'completed') {
+    return todosCopy.filter(todo => todo.completed === true);
+  }
+
+  return todosCopy;
+}
 
 export const App: React.FC = () => {
+  const [query, setQuery] = useState('');
+  const [todos, setTodos] = useState<Todo[]>([]);
+  const [filteredTodos, setFilteredTodos] = useState<Todo[]>([]);
+  const [errorMessage, setErrorMessage] = useState('');
+  const currentList = useRef('all');
+  const areAllCompleted = todos.every(todo => todo.completed);
+  const isAnyCompleted = todos.some(todo => todo.completed);
+  const noTodos = todos.length === 0;
+  const activeTodos = todos.filter(todo => !todo.completed);
+
+  useEffect(() => {
+    getTodos()
+      .then(response => {
+        setTodos(response);
+        setFilteredTodos(response);
+      })
+      .catch(() => setErrorMessage('Unable to load todos'));
+
+    setTimeout(() => setErrorMessage(''), 3000);
+  }, []);
+
+  const handleFiltration = (
+    e: React.MouseEvent<HTMLAnchorElement, MouseEvent>,
+  ) => {
+    const status = e.currentTarget.textContent?.toLowerCase();
+
+    setFilteredTodos(filterTodos(todos, status));
```

Comment:
you can just write in component
```
const filteredTodos = filterTodos(todos, status)
```

Reply:
Thanks. you're right, I remember doing it in previous task, but again got confused. Will fix it